src/drive/web/modules/services/components/SuggestionProvider.jsx

```diff
@@ -46,44 +47,30 @@ class SuggestionProvider extends React.Component {
     )
   }
 
-  // fetches pretty much all the files and preloads FuzzyPathSearch
+  // fetches pretty much all the files not trashed and preloads FuzzyPathSearch
   async indexFiles() {
     const { client } = this.props
-    // TODO: fix me
-    // eslint-disable-next-line no-async-promise-executor
-    return new Promise(async resolve => {
-      const resp = await cozy.client.fetchJSON(
-        'GET',
-        `/data/io.cozy.files/_all_docs?include_docs=true`
-      )
-      const files = resp.rows
-        // TODO: fix me
-        // eslint-disable-next-line no-prototype-builtins
-        .filter(row => !row.doc.hasOwnProperty('views'))
-        .map(row => ({ id: row.id, ...row.doc }))
 
-      const folders = files.filter(file => file.type === TYPE_DIRECTORY)
+    const files = await client.queryAll(buildSuggestionsQuery().definition)
```

Comment:
I'm wondering if this is not going to make the browser unresponsive. `client.query*` dispatch redux action, store result in a redux store, within the RAM and other. I'm not sure we need it here. 

What about calling `fetchJSON()` with the right path (the one you have inside the browser console) manually? Is it decrease the RAM consomption?

Reply:
- do you mean keeping the fetchJSON mechanism, already used in production? https://github.com/cozy/cozy-drive/blob/master/src/drive/web/modules/services/components/SuggestionProvider.jsx#L55

I am going to benchmark this.

About performance, 
I am following `about:performance` in Firefox and the `Task Manager` in Brave, without seeing a high consumption. 

I am going to work on the benchmark with lots of data to figure out, that problem.

Reply:
How many io.cozy.files do you have on your test env?

Reply:
100 000 files  now.

| Méthode         | Nb requêtes     | Data |Temps |
|--------------|-----------|------------|------------|
| fetchJSON | 8      | 12 Mo       | 9.7 s |
| queryAll      | **120**  | 8 Mo       | 3.2 s |

FetchJSON : les 8 requêtes (dont 6 liées à des notes) :
![Capture d’écran 2022-08-08 à 17 21 01](https://user-images.githubusercontent.com/8363334/183453843-ec8e5410-c6f9-4ef4-b2ab-525917131e38.png)


client.queryAll 2 cycles sur les 10 (exécutant à chaque fois 12 requêtes)
![Capture d’écran 2022-08-08 à 17 15 03](https://user-images.githubusercontent.com/8363334/183453909-8562a6e4-7e72-45ca-826c-961f3c102786.png)

Reply:
Tu donnes le fetchJSON ici sans l'optimisation que tu viens de faire à savoir de ne pas passer par all_docs. Ça ne fait pas sens. 

un truc du genre à tester : 
```
const req = await client.getStackClient.fetchJSON('POST', '/data/io.cozy.files', {le même selecteur que ton query})
while(req.next){
req.fetchJSON(...) //avec manipulation du bookmark à la main (comme le queryAll quoi)
}
```

A la fin tu dois avoir le même nombre de requête entre queryAll() et fetchJSON() car à la fin queryAll() fait un fetchJSON ! Et donc même nombre d'appels et même nombre de Mo transférés etc. 

Là où il va y avoir une différence c'est sur la RAM utilisée par l'application ! Car queryAll() va setter tout un store redux alors que fetchJSON ne va rien faire.

Reply:
Aussi, c'est via l'onglet Performances de tes dev tools qu'on verra le pic de CPU et de RAM ainsi que l'occupation du thread JS.

Reply:
after optims with paul:
![Capture d’écran 2022-08-08 à 18 43 24](https://user-images.githubusercontent.com/8363334/183469736-52eef42d-7d45-4b9b-b5db-85098c3d946d.png)

Reply:
another fetchJSON:
![Capture d’écran 2022-08-08 à 18 45 03](https://user-images.githubusercontent.com/8363334/183470037-a6d64519-5fd2-4648-9fd8-274377f20cb5.png)

Reply:
Idea: using a *fetchingAllJSONAndCreateIndex* method to handle a dynamic search + return result to display directly to the user. (what's about UserExperience)

Reply:
> Idea: using a fetchingAllJSONAndCreateIndex method to handle a dynamic search + return result to display directly to the user.

Can't we do something like:
```
const req = await client.collection('io.cozy.files').find(...)
```

to handle indexCreation and next attribute. 

We can create a DocumentCollection.findAll() method to handle the "full paginated result". @taratatach wants one too ;)

Reply:
https://github.com/cozy/cozy-client/pull/1219